include/oneapi/dpl/internal/dynamic_selection_impl/sycl_backend.h

```diff
@@ -159,17 +161,18 @@ class sycl_backend
     template <typename NativeUniverseVector>
     sycl_backend(const NativeUniverseVector& v)
     {
-        bool profiling = true;
         global_rank_.reserve(v.size());
         for (auto e : v)
         {
-            global_rank_.push_back(e);
-            if (!e.template has_property<sycl::property::queue::enable_profiling>())
-            {
-                profiling = false;
+            if(!e.get_device().has(sycl::aspect::ext_oneapi_queue_profiling_tag)){
+                if (!e.template has_property<sycl::property::queue::enable_profiling>()){
+                    auto prop_list = sycl::property_list{sycl::property::queue::enable_profiling()};
+                    auto e_tmp = sycl::queue{e.get_device(), prop_list};
```

Comment:
Should other properties also be copied from the original queue and then enable_profling added on top of those?